### ISSUE TITLE
I could not build/run examples

### ISSUE THREAD

**Issue by esn01** [PistonDevelopers/conrod#510]
Command: cargo run --verbose --example canvas

Error:
Compiling piston_window v0.7.0
     Running `rustc /home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs --crate-name piston_window --crate-type lib -g -C metadata=18d0a4700d934aac -C extra-filename=-18d0a4700d934aac --out-dir /home/slawek/src/conrod/target/debug/deps --emit=dep-info,link -L dependency=/home/slawek/src/conrod/target/debug/deps -L dependency=/home/slawek/src/conrod/target/debug/deps --extern gfx_graphics=/home/slawek/src/conrod/target/debug/deps/libgfx_graphics-fc6673b889f15738.rlib --extern shader_version=/home/slawek/src/conrod/target/debug/deps/libshader_version-471c178ebe98706a.rlib --extern gfx_device_gl=/home/slawek/src/conrod/target/debug/deps/libgfx_device_gl-74e546c0fd3bebc6.rlib --extern gfx=/home/slawek/src/conrod/target/debug/deps/libgfx-13346c6b03c0b277.rlib --extern glutin_window=/home/slawek/src/conrod/target/debug/deps/libglutin_window-22c87570a7b5b415.rlib --extern graphics=/home/slawek/src/conrod/target/debug/deps/libgraphics-34764b7e0ea47a00.rlib --extern piston=/home/slawek/src/conrod/target/debug/deps/libpiston-267d4dd847c23239.rlib -Awarnings -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu`
       Fresh piston2d-opengl_graphics v0.4.0
       Fresh gfx_device_gl v0.4.1
       Fresh pistoncore-glutin_window v0.4.0
       Fresh piston v0.3.0
/home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs:33:62: 33:90 error: wrong number of type arguments: expected 1, found 0 [E0243]
/home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs:33 pub type G2d<'a> = GfxGraphics<'a, gfx_device_gl::Resources, gfx_device_gl::CommandBuffer, gfx_device_gl::Output>;
                                                                                                                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs:33:62: 33:90 help: run `rustc --explain E0243` to see a detailed explanation
/home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs:130:39: 130:67 error: wrong number of type arguments: expected 1, found 0 [E0243]
/home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs:130             gfx_device_gl::Resources, gfx_device_gl::CommandBuffer,
                                                                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs:130:39: 130:67 help: run `rustc --explain E0243` to see a detailed explanation
error: aborting due to 2 previous errors
Could not compile `piston_window`.

Caused by:
  Process didn't exit successfully: `rustc /home/slawek/.cargo/registry/src/github.com-1ecc6299db9ec823/piston_window-0.7.0/src/lib.rs --crate-name piston_window --crate-type lib -g -C metadata=18d0a4700d934aac -C extra-filename=-18d0a4700d934aac --out-dir /home/slawek/src/conrod/target/debug/deps --emit=dep-info,link -L dependency=/home/slawek/src/conrod/target/debug/deps -L dependency=/home/slawek/src/conrod/target/debug/deps --extern gfx_graphics=/home/slawek/src/conrod/target/debug/deps/libgfx_graphics-fc6673b889f15738.rlib --extern shader_version=/home/slawek/src/conrod/target/debug/deps/libshader_version-471c178ebe98706a.rlib --extern gfx_device_gl=/home/slawek/src/conrod/target/debug/deps/libgfx_device_gl-74e546c0fd3bebc6.rlib --extern gfx=/home/slawek/src/conrod/target/debug/deps/libgfx-13346c6b03c0b277.rlib --extern glutin_window=/home/slawek/src/conrod/target/debug/deps/libglutin_window-22c87570a7b5b415.rlib --extern graphics=/home/slawek/src/conrod/target/debug/deps/libgraphics-34764b7e0ea47a00.rlib --extern piston=/home/slawek/src/conrod/target/debug/deps/libpiston-267d4dd847c23239.rlib -Awarnings -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu` (exit code: 101)

**Comment by pnkfelix** [comment]
see also https://github.com/gfx-rs/gfx/issues/786

**Comment by pnkfelix** [comment]
I was able to work around this by:

  1. Checking out the git version of `piston_window` (I did this in the parent directory of where I have put `conrod`, but I have seen other conventions that will work about where to put it.)
  2. Somewhere either in the `conrod/` directory, or on one of its parent directories, create a `.cargo/` subdirectory.  (Again, I did this in the parent directory of `conrod`.)
  3. Inside the newly created `.cargo/` subdirectory, create a `.cargo/config` file that shows the path to the locally checked out `piston_window` repository.
  4. Retry the `cargo build` of the example.

So, in command invocations:

```
% cd ..
% git clone git@github.com:PistonDevelopers/piston_window.git
% mkdir .cargo
% echo 'paths = ["piston_window"]' > .cargo/config
% cd conrod
% cargo build --example all_widgets
```

(The `canvas` example actually still does not build, because "the trait `event::generic_event::GenericEvent` is not implemented for the type `piston_window::PistonWindow`"; that might be related to the above workaround, or it might be an independent bug.)
